### PR TITLE
Feature/extraction

### PR DESCRIPTION
**Pull request by ShrawanBK**
Depends on https://github.com/idmc-labs/helix-server/pull/100

### CODE REVIEW

File: src/components/CountryMultiSelectInput/index.tsx [hunk 1]
```diff
@@ -40,30 +40,43 @@ type SelectInputProps<
     K,
     CountryOption,
     Def,
-    'onSearchValueChange' | 'searchOptions' | 'searchOptionsShownInitially' | 'optionsPending' | 'keySelector' | 'labelSelector'
->;
+    'onSearchValueChange' | 'searchOptions' | 'searchOptionsShownInitially' | 'optionsPending' | 'keySelector' | 'labelSelector' | 'regions'
```
Comment (tnagorra) [inline src/components/CountryMultiSelectInput/index.tsx:43]: You most probably don't need to add `'regions'` here

File: src/components/ExtractionQuery/SavedQueryList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            page,
+        });
+    }, [queryListFilters, queryListFilters]);
+
+    const onItemsPerPageChange = useCallback((pageSize) => {
+        setQueryListFilters({
+            ...queryListFilters,
+            pageSize,
+        });
+    }, [queryListFilters, queryListFilters]);
```
Comment (tnagorra) [inline src/components/ExtractionQuery/SavedQueryList/index.tsx:64]: Missing dependencies

File: src/components/ExtractionQuery/SavedQueryList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            search: text,
+        });
+    }, [queryListFilters, queryListFilters]);
+
+    const onActivePageChange = useCallback((page) => {
+        setQueryListFilters({
+            ...queryListFilters,
+            page,
+        });
+    }, [queryListFilters, queryListFilters]);
```
Comment (tnagorra) [inline src/components/ExtractionQuery/SavedQueryList/index.tsx:57]: Missing dependencies

File: src/components/ExtractionQuery/SavedQueryList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        queryListFilters,
+        setQueryListFilters,
+    } = props;
+
+    const onChangeSearchText = useCallback((text) => {
+        setQueryListFilters({
+            ...queryListFilters,
+            search: text,
+        });
+    }, [queryListFilters, queryListFilters]);
```
Comment (tnagorra) [inline src/components/ExtractionQuery/SavedQueryList/index.tsx:50]: Missing dependencies

File: src/components/ExtractionQuery/SavedQueryList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        onRefetchQueries,
+        queryListFilters,
+        setQueryListFilters,
+    } = props;
+
+    const onChangeSearchText = useCallback((text) => {
+        setQueryListFilters({
+            ...queryListFilters,
+            search: text,
+        });
```
Comment (tnagorra) [inline src/components/ExtractionQuery/SavedQueryList/index.tsx:49]: Prefer setQueryListFilters((oldFilter) => newFilter)) approach OR handle the mutation on parent.

File: src/components/ExtractionQuery/SavedQueryList/index.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    }, [queryListFilters, queryListFilters]);
+
+    const onActivePageChange = useCallback((page) => {
+        setQueryListFilters({
+            ...queryListFilters,
+            page,
+        });
```
Comment (tnagorra) [inline src/components/ExtractionQuery/SavedQueryList/index.tsx:56]: Prefer setQueryListFilters((oldFilter) => newFilter)) approach OR handle the mutation on parent.

File: src/components/ExtractionQuery/SavedQueryList/index.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    }, [queryListFilters, queryListFilters]);
+
+    const onItemsPerPageChange = useCallback((pageSize) => {
+        setQueryListFilters({
+            ...queryListFilters,
+            pageSize,
+        });
```
Comment (tnagorra) [inline src/components/ExtractionQuery/SavedQueryList/index.tsx:63]: Prefer setQueryListFilters((oldFilter) => newFilter)) approach OR handle the mutation on parent.

File: src/components/ExtractionQuery/ExtractionEntriesTable/index.tsx [hunk 8]
```diff
@@ -0,0 +1,359 @@
+import React, { useMemo, useState, useCallback, useContext } from 'react';
```
Comment (tnagorra) [inline src/components/ExtractionQuery/ExtractionEntriesTable/index.tsx:1]: Why not re-use EntriesTable?

File: src/components/ExtractionQuery/NewExtractionQuery/index.tsx [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+                },
+            });
+        }
+    }, [createExtraction, updateExtraction]);
+
+    const loading = createLoading || updateLoading || extractionQueryLoading;
+    const errored = !!extractionDataError;
+    const disabled = loading || errored;
+
+    const { user } = useContext(DomainContext);
```
Comment (tnagorra) [inline src/components/ExtractionQuery/NewExtractionQuery/index.tsx:297]: User is unused

File: src/components/ExtractionQuery/NewExtractionQuery/index.tsx [hunk 10]
```diff
@@ -0,0 +1,443 @@
...
+    // TODO: Permission based action on extraction query
```
Comment (tnagorra) [inline src/components/ExtractionQuery/NewExtractionQuery/index.tsx:298]: What do you mean by  "permission based action" ? Create/Edit/Delete?

File: src/components/ExtractionQuery/NewExtractionQuery/index.tsx [hunk 11]
```diff
@@ -0,0 +1,443 @@
...
+                    Generate
+                </Button>
+                <Button
+                    type="submit"
+                    name={undefined}
+                    disabled={disabled || pristine}
+                    variant="primary"
+                    className={styles.button}
+                >
+                    {value.id ? 'Update Query' : 'Save Query'}
```
Comment (tnagorra) [inline src/components/ExtractionQuery/NewExtractionQuery/index.tsx:436]:
```suggestion
                    Save Query
```

File: src/components/ExtractionQuery/SavedQueryList/QueryItem/index.tsx [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+function QueryItem(props: QueryItemProps) {
+    const {
+        query,
+        onRefetchQueries,
+        className,
+    } = props;
+
+    const { user } = useContext(DomainContext);
```
Comment (tnagorra) [inline src/components/ExtractionQuery/SavedQueryList/QueryItem/index.tsx:41]: User is unused

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/components/ExtractionQuery/ExtractionEntriesTable/index.tsx [hunk 13]
```diff
@@ -0,0 +1,359 @@
...
+            contentClassName={styles.content}
+            // headerActions={(
+            //     <TextInput
+            //         icons={<IoIosSearch />}
+            //         name="search"
+            //         value={searchText}
+            //         placeholder="Search"
+            //         onChange={setSearchText}
+            //     />
+            // )}
```
Comment (tnagorra) [inline src/components/ExtractionQuery/ExtractionEntriesTable/index.tsx:331]: Just remove this

File: src/components/ExtractionQuery/SavedFiltersList/QueryItem/index.tsx [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+                const { deleteExtraction: deleteExtractionRes } = response;
+                if (!deleteExtractionRes) {
+                    return;
+                }
+                const { errors, result } = deleteExtractionRes;
+                if (errors) {
+                    notify({ children: 'Sorry, the query could not be deleted!' });
+                }
+                if (result) {
+                    notify({ children: 'Extraction Query deleted successfully!' });
```
Comment (tnagorra) [inline src/components/ExtractionQuery/SavedFiltersList/QueryItem/index.tsx:60]:
```suggestion
                    notify({ children: 'Query deleted successfully!' });
```

File: src/components/ExtractionQuery/SavedFiltersList/index.tsx [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+                        query={query}
+                        onRefetchQueries={onRefetchQueries}
+                    />
+                </div>
+            ))}
+
+            {loading && <Loading absolute />}
+            {!loading && totalQueryCount <= 0 && (
+                <Message
+                    message="No extraction query found."
```
Comment (tnagorra) [inline src/components/ExtractionQuery/SavedFiltersList/index.tsx:106]:
```suggestion
                    message="No saved query found."
```

File: src/components/FigureTagMultiSelectInput/index.tsx [hunk 16]
```diff
@@ -10,12 +10,20 @@ import {
 } from '@togglecorp/toggle-ui';
 
 import useDebouncedValue from '#hooks/useDebouncedValue';
+<<<<<<< HEAD
 import { GetFigureTagListQuery, GetFigureTagListQueryVariables } from '#generated/types';
+=======
+import { GetFigureTagsQuery, GetFigureTagsQueryVariables } from '#generated/types';
+>>>>>>> b5a1202... CRUD on Extraction Query
```
Comment (tnagorra) [inline src/components/FigureTagMultiSelectInput/index.tsx:17]: Conflict markers

File: src/components/FigureTagMultiSelectInput/index.tsx [hunk 17]
```diff
@@ -10,12 +10,20 @@ import {
...
 
 import styles from './styles.css';
 
 const FIGURE_TAGS = gql`
+<<<<<<< HEAD
     query GetFigureTagList($search: String){
+=======
+    query GetFigureTags($search: String){
+>>>>>>> b5a1202... CRUD on Extraction Query
```
Comment (tnagorra) [inline src/components/FigureTagMultiSelectInput/index.tsx:26]: Conflict markers

File: src/components/FigureTagMultiSelectInput/index.tsx [hunk 18]
```diff
@@ -25,7 +33,11 @@ const FIGURE_TAGS = gql`
     }
 `;
 
+<<<<<<< HEAD
 export type FigureTagOption = NonNullable<NonNullable<GetFigureTagListQuery['figureTagList']>['results']>[number];
+=======
+export type FigureTagOption = NonNullable<NonNullable<GetFigureTagsQuery['figureTagList']>['results']>[number];
+>>>>>>> b5a1202... CRUD on Extraction Query
```
Comment (tnagorra) [inline src/components/FigureTagMultiSelectInput/index.tsx:40]: Conflict markers

File: src/components/FigureTagMultiSelectInput/index.tsx [hunk 19]
```diff
@@ -52,7 +64,11 @@ function FigureTagMultiSelectInput<K extends string>(props: SelectInputProps<K>)
     const debouncedSearchText = useDebouncedValue(searchText);
 
     const searchVariable = useMemo(
+<<<<<<< HEAD
         (): GetFigureTagListQueryVariables | undefined => (
+=======
+        (): GetFigureTagsQueryVariables | undefined => (
+>>>>>>> b5a1202... CRUD on Extraction Query
```
Comment (tnagorra) [inline src/components/FigureTagMultiSelectInput/index.tsx:71]: Conflict markers

File: src/components/FigureTagMultiSelectInput/index.tsx [hunk 20]
```diff
@@ -61,7 +77,11 @@ function FigureTagMultiSelectInput<K extends string>(props: SelectInputProps<K>)
     const {
         loading,
         data,
+<<<<<<< HEAD
     } = useQuery<GetFigureTagListQuery>(FIGURE_TAGS, {
+=======
+    } = useQuery<GetFigureTagsQuery>(FIGURE_TAGS, {
+>>>>>>> b5a1202... CRUD on Extraction Query
```
Comment (tnagorra) [inline src/components/FigureTagMultiSelectInput/index.tsx:84]: Conflict markers

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**